src/v/kafka/protocol/batch_consumer.h

```diff
@@ -55,11 +48,6 @@ class kafka_batch_serializer {
         if (unlikely(record_count_ == 0)) {
             _base_offset = batch.base_offset();
         }
-        if (unlikely(
-              !_first_tx_batch_offset
-              && batch.header().attrs.is_transactional())) {
-            _first_tx_batch_offset = batch.base_offset();
-        }
```

Comment:
> No longer needed as we retain abort markers even after
compaction.

Is this a requirement even for data that was written before this change?

Reply:
That is a great observation, you are right this part of the change should be retained for the mixed mode data.